packages/next/plugins/with-nx.ts

```diff
@@ -210,12 +213,11 @@ function withNx(
       const userWebpackConfig = nextConfig.webpack;
 
       const { createWebpackConfig } = require('@nx/next/src/utils/config');
+      // If we have file replacements or assets, inside of the next config we pass '.' as the workspaceRoot so that the paths are relative to the project.
       nextConfig.webpack = (a, b) =>
         createWebpackConfig(
-          workspaceRoot,
-          projectDirectory,
-          options.fileReplacements,
-          options.assets
+          _nextConfig.nx?.fileReplacements ? '.' : workspaceRoot,
```

Comment:
For `@nx/next:build` executor usages, we still want to ensure paths relative to workspace root continue to work. I don't think we want to change how `createWebpackConfig` functions, but rather we should normalize the paths from `fileReplacements` and `assets` prior to calling `createWebpackConfig`.

Reply:
make `.` an absolute paths

Reply:
Shouldn't `path.resolve` which is inside `createWebpackConfig` already ensures that the path we pass is absolute: https://nodejs.org/api/path.html#pathresolvepaths ?

I passed the `projectDirectory`  instead of `.` which is better for readability